### ISSUE TITLE
[HOLD for #17387] [$1000]  Inconsistent Error Messages Displayed for Invalid Email Addresses During Sign Up Process

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21110]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to the Staging website.
2. Enter various invalid email addresses, such as jane.doe@gmail..com, [jane.doe@gmail.com](mailto:jane.doe@gmail.com)@example.com, and jane.doe@com.
3. Note that the same error message is displayed for all invalid email addresses except for jane.doe@-gmail.com.
4. This inconsistency in error handling can cause confusion for users.

## Expected Result:
The app should display the same error message for all invalid email addresses to ensure consistency in error handling.

The message for any incorrectly formatted emails should be "The email entered is invalid. Please fix the format and try again"

## Actual Result:
The app displays different error messages for this kind of email (eg. jane.doe@-gmail.com), leading to inconsistency in error handling.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.29-3

**Reproducible in staging?:** y

**Reproducible in production?:** y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

[screen-capture (94) (1).webm](https://github.com/Expensify/App/assets/93399543/a8e7db3c-a47d-4693-a7a4-6340d3f52f7e)

https://github.com/Expensify/App/assets/93399543/93041f1f-b27f-4b9f-bdf6-8e1b6602bacc

**Expensify/Expensify Issue URL:**

**Issue reported by:** @tewodrosGirmaA

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686679747437349

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0130df66afd4333dd5</li>
        <li>Upwork Job ID: 1671627355625070592</li>
        <li>Last Price Increase: 2023-06-28</li>
    </ul>
</details>

**Comment by payal-lathidadiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistent Error Messages Displayed for Invalid Email Addresses During Sign Up Process

### What is the root cause of that problem?
Front end (client-side) validation picks validation message from `loginForm.error.invalidFormatEmailLogin` [here](https://github.com/Expensify/App/blob/main/src/pages/signin/LoginForm.js#LL151C2-L158C10) 

below localized text from en.js which is for client side validation
```
invalidFormatEmailLogin: 'The email entered is invalid. Please fix the format and try again.',
```
While different error message returned from API when email validation is failed from backend
<img width="828" alt="image" src="https://github.com/Expensify/App/assets/130229379/7da11c67-8a9a-4a6e-af5e-7094d493120a">

This caused inconsistency in error message.

### What changes do you think we should make in order to solve the problem?
Change either front end or backend error message to have same message for email validation failure.

### What alternative solutions did you explore? (Optional)

**Comment by mohamedghan** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue.
Getting different error messages for fairly the same error
### What is the root cause of that problem?
the root is definitely the frontend given that in the case of the address jane.doe@-gmail.com the frontend accepts the address and sends it to the backend. however in the case of the address jane.doe@com, the frontend blocks the request from being sent to the server and shows an error  message generated from the frontend

![image](https://github.com/Expensify/App/assets/37912741/24c9fba3-db8a-4c54-b6bb-4ddee3f02590)

![image](https://github.com/Expensify/App/assets/37912741/d150aa8f-2c88-44cc-8620-d77aaac028ad)


### What changes do you think we should make in order to solve the problem?

changing the frontend so that it blocks both wrong emails jane.doe@com and  jane.doe@-gmail.com from being sent to the backend 
### What alternative solutions did you explore? (Optional)

**Comment by mohamedghan** [comment]
Contributor details
Your Expensify account email: ghanmimohamed001@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~010df8f0e38f174768

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
For invalid emails different message gets displayed to the user

### What is the root cause of that problem?
dash/hyphen at the start/end of the domain we are allowing in the frontend as per the expensify-common regex [here](https://github.com/Expensify/expensify-common/blob/f9985d572c772757885ff59ea5c82b1050c6b3f7/lib/CONST.jsx#L3) whereas backend is throwing an error for those emails. So for these emails we are getting the error message from the api and that is causing the inconsistent error message for invalid emails.

### What changes do you think we should make in order to solve the problem?
We need to update the regex which should not allow hyphen(-) at the start/end of the domain name which will be consistent with the backend regex.

Instead of 
```
const EMAIL_BASE_REGEX = "([\\w\\-\\+\\'#]+(?:\\.[\\w\\-\\'\\+]+)*@(?:[\\w\\-]+\\.)+[a-z]{2,})";
```
 we can update to 
```
const EMAIL_BASE_REGEX = "([\\w\\-\\+\\'#]+(?:\\.[\\w\\-\\'\\+]+)*@(?:[\\w]+(?:\\-?[\\w]+)*\\.)+[a-z]{2,})";
```

**Comment by MitchExpensify** [comment]
Updated the OP to call out the expected behavior: We want this message to show in all these cases: "The email entered is invalid. Please fix the format and try again"

**Comment by payal-lathidadiya** [comment]
## Proposal
Post clarification on expected behavior/OP;

Updates to my proposal [here](https://github.com/Expensify/App/issues/21110#issuecomment-1598640296)

### What changes do you think we should make in order to solve the problem?
Changing regular expression might not help here as backend uses `FILTER_VALIDATE_EMAIL` which has different logic than what we have in `EMAIL_BASE_REGEX`. 

Updating `EMAIL_BASE_REGEX` to consider email with hyphen(-) as domain prefix as invalid may solve only current use case, but there could be other scenarios where frontend validation is differing from backend logic.

So, we should update backend email validation failure message as below
"**The email entered is invalid. Please fix the format and try again**"

### What alternative solutions did you explore? (Optional)

**Comment by MitchExpensify** [comment]
How does @payal-lathidadiya's proposal look to you now @0xmiroslav

**Comment by MitchExpensify** [comment]
Bump @0xmiroslav 😄

**Comment by MitchExpensify** [comment]
Awesome, sounds likes a plan! Thanks @0xmiroslav

**Comment by MitchExpensify** [comment]
Not overdue, switching to weekly while we wait on https://github.com/Expensify/App/issues/17387

**Comment by MitchExpensify** [comment]
Still waiting on https://github.com/Expensify/App/issues/17387

**Comment by MitchExpensify** [comment]
Still waiting on https://github.com/Expensify/App/issues/17387

**Comment by MitchExpensify** [comment]
https://github.com/Expensify/App/issues/17387 is getting closer! _Soon_

**Comment by MitchExpensify** [comment]
https://github.com/Expensify/App/issues/17387 is pending review - Still on hold

**Comment by MitchExpensify** [comment]
Fixed in a separate issue that had this covered before this was reported here - closing :

<img width="401" alt="image" src="https://github.com/Expensify/App/assets/36425901/11e6fa3e-8d34-4d38-a79f-84f2b107ebc4">

**Comment by tewodrosGirmaA** [comment]
Good morning @MitchExpensify I am not eligible for bug bountey?

**Comment by MitchExpensify** [comment]
The issue this was ultimately resolved because of, and pointed out the general inconsistency, was from April and you reported in June so figured the reporting bonus did not apply here @tewodrosGirmaA
